airflow/models/baseoperator.py

```diff
@@ -1245,11 +1245,21 @@ def dag(self) -> DAG:  # type: ignore[override]
     @dag.setter
     def dag(self, dag: DAG | None):
         """Operators can be assigned to one DAG, one time. Repeat assignments to that same DAG are ok."""
-        from airflow.models.dag import DAG
-
         if dag is None:
             self._dag = None
             return
+
+        # if already set to elided, then just set and exit
+        if self._dag is ELIDED_DAG:
```

Comment:
Can we change the name to `suppressed` ? I had to look it up in dictionary and `elided` is rather sophisticated word that is likely not in vocabulary of many non-native people.

Reply:
Absolutely. It’s a weird word. Very unusual word.

Reply:
Another also more common and matching term would be "omit"? Suppressed sounds like hiding something?
```suggestion
        if self._dag is OMITTED_DAG:
```

Reply:
or maybe approaching it from other direction `VISITED_DAG` would be best...

Reply:
yeah, i didn't land on ELIDE just for the sake of being esoteric. it's actually a very specific thing and there was not a word could find that described it better.  

the sense in which i use it is like, it's intentionally removed (but you can't say "deleted" since that implies the dag is deleted).

maybe it's not such a bad thing to have to look up the word :)

anyway, omit is close.  how about simply `MISSING_DAG`????

Reply:
I think VISITED or ALREADY_SERIALIZED is better name - it comes from the other direction - describes what happens with the dag and based on that, serialization code decides what to do. tthe ELIDE or MISSING is a bit "future" or "crystal ball" approach. How do you know you should elide it a the moment you add it to the kist of dags ? Turning that around and naming it as "VISITED" or "ALREADY_SERIALIZED" is IMHO better. The code will read as 

```
if THIS NODE WAS VISITED ALREADY:
   skip it
``

Rather than:

```
if SHOULD_I_SKIP_IT:
   skip it
```

The second case is kinda duplication and "future looking".

Reply:
I think I get what you're saying.  The thing that doesn't feel right is that we're saying 

```
if task.dag is HAS_BEEN_SERIALIZED
```

which reads like the _dag_ has been serialized, but that's not quite what it's saying; in reality, the _task_ has been serialized (or visited if you like). and the dag obj on the task was _removed_. 

"visited" seems a bit insufficient because it doesn't really tell you what happened when you visited.

How about this

```
ATTRIBUTE_REMOVED
```

I think "attribute removed" is good because it's very clear, and tells the truth.

Reply:
BTW, this sentinel is private so, not user-facing, so of less consequence

Reply:
resolving for now and you can start new convo after checking out the change if desired